lib/Common/DataStructures/DictionaryEntry.h

```diff
@@ -6,81 +6,77 @@
 
 namespace JsUtil
 {
-    template <class TValue>
-    class BaseValueEntry
+    // Packing matters because we make so many dictionary entries.
+    // The int pointing to the next item in the list may be included
+    // either after the value or after the key, depending on which
+    // packs better.
+
+    namespace details
```

Comment:
nit: Most of our namespaces are PascalCased, and I think `details` could get a more descriptive name -- or, since its not used elsewhere, I would actually suggest simply using an anonymous namespace.